### PR TITLE
Add OrderList performance transaction

### LINKED ISSUE #7063: [Orders Performance] Introduce loading waiting time metric to the Orders list view

### PR DESCRIPTION
**Pull request by wzieba**
Closes: #

### Description

This PR adds performance tracking for the loading list of orders.

### Testing instructions

1. Open the app
2. Go to list of orders
3. Verify that after a few seconds, you can see a new transaction (with HTTP, db and UI spans) in Sentry Performance Monitoring dashboard. Transaction name: `OrderList` (don't forget to sort transactions as "recent first")

### Images/gif

image

- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListTransactionLauncher.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    override fun onStateChanged(source: LifecycleOwner, event: Lifecycle.Event) {
+        when (event) {
+            Lifecycle.Event.ON_CREATE -> {
+                performanceTransactionId =
+                    performanceTransactionRepository.startTransaction(TRANSACTION_NAME, TransactionOperation.UI_LOAD)
+            }
+            Lifecycle.Event.ON_STOP -> {
+                performanceTransactionId?.let {
+                    performanceTransactionRepository.finishTransaction(it, TransactionStatus.ABORTED)
+                }
```
Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListTransactionLauncher.kt:68]: We abort transactions for `ON_STOP` as, suprisingly, `ON_PAUSE` is not invoked if we change tabs very quickly. 

I forgot to prepare evidence but you can test if with some logger in line 64 and compare events when quickly changing bottom navigation tabs, vs. doing it slowly.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListTransactionLauncher.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                    performanceTransactionId?.let {
+                        performanceTransactionRepository.finishTransaction(it, TransactionStatus.SUCCESSFUL)
+                    }
+                }
+            }
+        }
+    }
+
+    private enum class Conditions {
+        FIRST_LIST_FETCHED,
```
Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListTransactionLauncher.kt:46]: As decided, this is the only condition (internal: pe5pgL-hq-p2)

Comment (ThomazFB) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListTransactionLauncher.kt:46]: Nice to see the follow-up on the discussion! But it's also great that we can easily increase the conditions in the future if we want. Nice work!

**Review by ThomazFB — APPROVED** [review]
Working like a charm! Looks good to me! 
image

**Pull request merged**